zulip/zulip/__init__.py

```diff
@@ -617,17 +617,17 @@ def call_on_each_event(
         callback: Callable[[Dict[str, Any]], None],
         event_types: Optional[List[str]] = None,
         narrow: Optional[List[List[str]]] = None,
+        kwargs: Dict[str, object] = {},
```

Comment:
It’s dangerous to specify a mutable type for a parameter with a default argument, because the default argument is shared between all callers; use `Mapping`, not `Dict`.

However, enclosing these extra arguments in a dictionary is inconsistent with the `register` API. I think we should use `**kwargs`.